### PR TITLE
Add git pull support for Visual Studio 2017 and 2019

### LINKED ISSUE #2: Add support for versions of Visual studio before 2019
The current VSIX only supports Visual Studio 2019:
https://github.com/madskristensen/GitPull/blob/master/src/source.extension.vsixmanifest#L13

Could we make it work for earlier versions as well? 😄

### PR DESCRIPTION
**Pull request by jcansdale**
The aim of this PR is to use Team Explorer to perform a git pull after Hub has synced all branches. This is complicated by Team Explorer not making this functionality easily available. 😉 

### What this PR does

- Add support for Visual Studio 2017 and 2019
- Target Visual Studio 2017 15.6 and above (`FileAndForget` was added in 15.6)
- Sync all branches using `Hub`
- Send output to existing `Source Control - Git` output pane
- Send % output to status bar (Hub exposes this is standard Error)
- Use `Team Explorer - Synchronization` page to `pull` the current branch
- Uses [IgnoresAccessChecksToGenerator](https://github.com/aelij/IgnoresAccessChecksToGenerator) to access internal Team Explorer classes (e.g. `GitCommitsPageViewModel`)

### To do

- [x] Hub creates lots of noise when enumerating and downloading objects

E.g:
```
remote: Enumerating objects: 101, done.        
remote: Counting objects:   0% (1/101)           
remote: Counting objects:   1% (2/101)           
remote: Counting objects:   2% (3/101)           
remote: Counting objects:   3% (4/101)           
remote: Counting objects:   4% (5/101)           
remote: Counting objects:   5% (6/101)           
remote: Counting objects:   6% (7/101)           
remote: Counting objects:   7% (8/101)           
remote: Counting objects:   8% (9/101)           
remote: Counting objects:   9% (10/101)           
remote: Counting objects:  10% (11/101)   
```

Maybe this could be recognized and redirected to the status bar?

- [ ] Lots of build warnings

E.g.

```
  GitPull.Services -> C:\projects\gitpull\src\GitPull.Services\bin\Release\net46\GitPull.Services.dll
C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\MSBuild\15.0\Bin\Microsoft.Common.CurrentVersion.targets(2110,5): warning MSB3277: Found conflicts between different versions of "Microsoft.VisualStudio.Threading" that could not be resolved.  These reference conflicts are listed in the build log when log verbosity is set to detailed. [C:\projects\gitpull\src\GitPull.Services.15.0\GitPull.Services.15.0.csproj]
C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\MSBuild\15.0\Bin\Microsoft.Common.CurrentVersion.targets(2110,5): warning MSB3277: Found conflicts between different versions of "Newtonsoft.Json" that could not be resolved.  These reference conflicts are listed in the build log when log verbosity is set to detailed. 
```
https://ci.appveyor.com/project/madskristensen/gitpull/builds/21281211#L37

@madskristensen do you know if there is any way to fix these? ☝️ 

Fixes #2

### CODE REVIEW

File: src/GitPull/source.extension.vsixmanifest [hunk 1]
```diff
@@ -10,7 +10,7 @@
         <Tags>git</Tags>
     </Metadata>
     <Installation>
-        <InstallationTarget Id="Microsoft.VisualStudio.Community" Version="[16.0, 17.0)" />
+        <InstallationTarget Id="Microsoft.VisualStudio.Community" Version="[15.6, 17.0)" />
```
Comment (jcansdale) [inline src/GitPull/source.extension.vsixmanifest:13]: It looks like `15.6` doesn't mean what I thought it meant. This version won't install on VS 2017 15.9. 😕 

@madskristensen What should this version be if I want it to install on 15.6 or above?

Comment (madskristensen) [reply src/GitPull/source.extension.vsixmanifest:13]: 15.6 is "15.0.27428". I use this website to find out https://docs.microsoft.com/en-us/visualstudio/install/visual-studio-build-numbers-and-release-dates?view=vs-2017

**Comment by madskristensen** [comment]
Let me know when I should merge this PR

**Comment by madskristensen** [comment]
Actually, let me just make you contributor on this

**Comment by tekkenxpert** [comment]
Hi all,
Is this extension compatible with VS2017? I'm getting the below error

Capture

**Comment by madskristensen** [comment]
@jcansdale is this ready for merge?

**Comment by jcansdale** [comment]
Hi @madskristensen 👋

In hindsight I think this was too risky. It was accessing internal functionality IIRC. 😢 

Alas, I don't spend much time inside VS these days. I'm going to close this PR as an experiment.

**Comment by madskristensen** [comment]
ok

**Comment by tekkenxpert** [comment]
Hi mads/jcansdale,

Is this supported in VS 2019, or are you dropping this completely?

**Comment by jcansdale** [comment]
@tekkenxpert,

> Is this supported in VS 2019, or are you dropping this completely?

It was using a trick to access an internal API, see the `IgnoresAccessChecksToGenerator` package [here](https://github.com/madskristensen/GitPull/pull/3/files#diff-516ffc3c6c49899b41fb396900983359R14) and [here](https://github.com/madskristensen/GitPull/pull/3/files#diff-602d18f0a952516cc407feacef5d0854R34-R36). This worked but was maybe a little too sketchy to be supportable. 😉 

I know the Visual Studio team are doing a complete revamp of their Git support (see recent Preview versions of VS). If this is a feature you really want, this might not be a bad time to ask about it. I think there is similar functionality in VS Code which could help your case.

At the moment I'm a somewhat reluctant Mac user, so I don't get much time to hack on VS. 😢

**Pull request closed without merging**